services/configuration/src/api.rs

```diff
@@ -230,12 +231,7 @@ impl PublicApi {
     }
 
     fn retrieve_filter(&self, request: &mut Request) -> Result<Filter, ApiError> {
-        let previous_cfg_hash: Option<Hash> = self.optional_param(request, "previous_cfg_hash")?;
-        let actual_from: Option<Height> = self.optional_param(request, "actual_from")?;
-        Ok(Filter {
-            previous_cfg_hash,
-            actual_from,
-        })
+        self.parse(request).or(Ok(Filter::default()))
```

Comment:
Are you sure this is right? As written now, his method *never* fails, including the case when the body is malformed or non-existent.